### ISSUE TITLE
Improve docs on Caching

### ISSUE THREAD

**Issue by shoooe** [solidjs/solid-docs#966]
### 📋 Explain your issue

A common document page in other frameworks is caching. Since SolidJS just uses Nitro, caching can be easily implemented via the `routeRules` setting as described here: https://stackoverflow.com/q/79223996/493122

Would this be worth adding to the docs?

**Comment by brenelz** [comment]
Could maybe add a note here https://docs.solidjs.com/solid-start/reference/config/define-config, but I'm sure where the docs stand on nitro specific things when you could just look at things over there

**Comment by atilafassina** [comment]
Yeah, we try not to document Nitro directly (because it's a slippery slope, soon we end up documenting H3, Vite, etc...) - but I believe a note to that would be helpful.

We could also add a Caching section to our Learn guide for start where we explain our deduplication system _and_ how to leverage Nitro caching, _and_ how to play with HTTP headers for Max-Age, SWR, etc on Server Functions (tapping into the request event and so on) - that would be super useful.

## suggested action items
- [ ] Add a explicit note about caching on the "configuring Nitro" section
- [ ] Create a section on how to handle cache within Server Functions in `SolidStart / Guides / Caching`
